docs/source/contributing.rst

```diff
@@ -1,12 +1,223 @@
+
+Clone the repository from the link in the master section of the doc. You now have a copy of your repository's master branch available to you. When
+you have an idea of what you'd like to contribute, create a new branch before
+starting work.
+
+Your new branch contains a copy of the content of master. Make your additions
+and edits now, they will only affect the new branch. When done add
+and commit the files.
+
+You now have a branch with your contribution, but you haven't contributed until
+the code makes it into the main repository. This involves several steps. First,
+any changes others have made in the team's repository need to be merged into
+your code.
+
+.. tip::
+
+    If the work done on the branch stretches over a long period, it is good practice to frequently merge the master branch into your personal branch.
+
+If there are any conflics Git can't resolve automatically, it is
+your job to `resolve <https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line>`_ those errors. By merging changes into your contribution, rather than the other way around, you ensure the act of bringing your code into the team's repository will go smoothly. This helps when another member of the team reviews your code as well.
+
+Now that you have a merged branch, you should push the branch to your GitHub.
+From GitHub, you can make a `pull request <https://help.github.com/articles/using-pull-requests/>`_ from your repository against the team's repository.
+
+.. note::
+
+    A link to a edit a pull request is also automatically generated when the branch is pushed in the command line. You can use this link to create the pull request.
+
+Pull Requests (PR)
+~~~~~~~~~~~~~~~~~
+Pull requests are how we are able to review each others changes to the master branch. It helps you communicate your changes to ``ros2``. You can choose to create a pull request that is ready for review or draft a pull request. Draft PRs must be marked as "ready for review" before they can be merged into the master branch.
+
+Once your PR is ready for review, assign an older team member to review the PR. This will notify them that you are ready to have your contribution reviewed. Requirements for pull request standards are listed in several sections below. The team member may ask that you fix or touch up some things before the request is accepted. This is normal and common. Once the pull request meets standards, the older member will approve it, and your changes can be merged (provided they pass the CI checks, which are explained below).
+
+Keep in mind, you can have several branches at once. If you need to fix a bug
+for an existing contribution while working on a new one, you should checkout
+the master branch, and then create a new branch named bug fix. It is
+critically important that a pull request only addresses one thing at a time. If it does not, the request will not be accepted until you have properly separated the items you have worked on.
```

Comment:
If you want to enforce this, updating out PR templates would help with that. [here ](https://github.com/UBC-Thunderbots/Software/blob/master/.github/PULL_REQUEST_TEMPLATE.md)is a nice example of a template from Thunderbots

Reply:
They haven't been updated in 2 years, so I think it is time to refresh them.